### ISSUE TITLE
Has anyone figured out how to change resolution?

### ISSUE THREAD

**Issue by ClashSAN** [bes-dev/stable_diffusion.openvino#101]
openvino still runs faster than diffusers-onnxruntime cpu, but changing size and IR format model conversion from onnx still is not understood.

We can add DDIM support and produce good results at 8-14 steps, since this outdated diffusers has ddim support. We could also maybe hack in DPM++ 2M too. for even lower steps.

Here are the enhancements: https://github.com/ClashSAN/portable_stable_diffusion.openvino

**Comment by RedAndr** [comment]
The latest diffusers version has Euler Ancestral. It's quite easy to incorporate it, it just needs some conversion code. It works even faster than DDIM and gives better results most of the time.

**Comment by ClashSAN** [comment]
@Seegee nono, it's unrelated to resolution.

It's been mentioned in other issues here the size is being hard-coded somewhere but I don't know enough to make the changes.

**Comment by RedAndr** [comment]
Actually, it's possible to change the resolution. But you would have to convert the model to another one. Or if you need flexible resolution, you can add dynamic axes during conversion. Although it makes the model much slower, about three times.

**Comment by ClashSAN** [comment]
@RedAndr are you referring to https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/225-stable-diffusion-text-to-image/225-stable-diffusion-text-to-image.ipynb converted model + inference being slower than the one from this repository? So maybe cpu onnx is still a better option?

**Comment by RedAndr** [comment]
@ClashSAN No, sorry, I meant the model converted with dynamic axes is slower.
`torch.onnx.export(dynamic_axes={"init_image": {0: "batch", 1: "channels", 2: "height", 3: "width"}})`
BTW, ONNX is also slower than OpenVINO IR.

**Comment by ClashSAN** [comment]
I didn't get https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/225-stable-diffusion-text-to-image/225-stable-diffusion-text-to-image.ipynb working yet on colab or locally but once I do I'll see if I can upload some of custom models to huggingface

**Comment by RedAndr** [comment]
Locally it works Ok, colab should too, but didn't try.

**Comment by ClashSAN** [comment]
@RedAndr got it working, it used a ton of ram to convert the unet. My 8gb+24swap couldn't convert this.

https://huggingface.co/ClashSAN/openjourney-openvino-IR/tree/main

when using different sizes, did the ram usage drop? enough to fit 8gb?

in my environment, i go from 16s/it (onnx) -> 13s/it (openvino)

same speeds as this repo

**Comment by RedAndr** [comment]
Very good! Yes, memory consumption depends on the resolution pretty much. But even with 128GB of RAM, I couldn't convert a model larger than 1024x768. I don't have a formula, you need to check yourself, which resolution could fit into your amount or memory.

**Comment by ClashSAN** [comment]
The ncnn [repo](https://github.com/EdVince/Stable-Diffusion-NCNN) is currently a tad bit slower (pc) 5.3s/it vs 5.6s/it but does dynamic shapes properly. It use 11.5gb of ram peak for me.

Then there's another https://github.com/fengwang/Stable-Diffusion-NCNN (less ram) I haven't tested.

@RedAndr ever thought about making a repo to guide users to use this? Openvino will provide speed improvements for intel hardware, and for users under 8gb, they can use onnx models, which my converted f16 model only takes an additional 2.5gb for 320x384 (good for phone users.)

Then, you can further quantize the model to get 2x speed. At 256x320, it takes only .8gb.

**Comment by RedAndr** [comment]
@ClashSAN Ok, here's my version of the converter: https://github.com/RedAndr/SD_PyTorch2ONNX
It has dynamic axes, so any resolution could be used. To have a constant resolution just comment lines with dynamic_axes.
Let me know if you have questions.
